### PR TITLE
Release 81.0.0

### PR DESCRIPTION
**Pull request by OGPoyraz**
See changelogs

### CODE REVIEW

File: packages/gas-fee-controller/CHANGELOG.md
Comment (OGPoyraz) [reply packages/gas-fee-controller/CHANGELOG.md:11]: Thanks, I will remove them all now

Comment (OGPoyraz) [reply packages/gas-fee-controller/CHANGELOG.md]: Fixed in https://github.com/MetaMask/core/pull/1747/commits/827211aea6fc76a50d38c6ed0f588adbb7d8f76e

File: packages/gas-fee-controller/CHANGELOG.md
Comment (OGPoyraz) [reply packages/gas-fee-controller/CHANGELOG.md]: All removed https://github.com/MetaMask/core/pull/1747/commits/827211aea6fc76a50d38c6ed0f588adbb7d8f76e

File: packages/gas-fee-controller/CHANGELOG.md
Comment (OGPoyraz) [reply packages/gas-fee-controller/CHANGELOG.md:11]: Added in https://github.com/MetaMask/core/pull/1747/commits/8d8422bd73bcce88cc73b8d7029f48fb1b9e20f5

Comment (OGPoyraz) [reply packages/gas-fee-controller/CHANGELOG.md:11]: Sure, I will keep this until getting some feedback from @adonesky1

File: packages/assets-controllers/CHANGELOG.md
Comment (adonesky1) [inline packages/assets-controllers/CHANGELOG.md:36]: I should have added one other piece here:
```suggestion
- `NftController.addNftVerifyOwnership`: Now accepts optional 3rd argument `networkClientId` which is used to fetch NFT metadata and determine by which chainId the added NFT should be stored in state. Also accepts optional 4th argument `source` used for metrics to identify the flow in which the NFT was added to the wallet.  ([#1698](https://github.com/MetaMask/core/pull/1698))
```

File: packages/gas-fee-controller/CHANGELOG.md
Comment (adonesky1) [reply packages/gas-fee-controller/CHANGELOG.md:11]: Ok https://github.com/MetaMask/core/pull/1748 was merged, so this should be rebased and the polling-controller version bumped to 0.0.1 in the gasFeeController package.json in this release.

Comment (adonesky1) [reply packages/gas-fee-controller/CHANGELOG.md:11]: I can do this just want to make sure someone else isn't in the middle of it. Let me know.

File: packages/gas-fee-controller/CHANGELOG.md
Comment (mcmire) [inline packages/gas-fee-controller/CHANGELOG.md:11]: The formatting looks off here, perhaps this should be:

``` suggestion
- **BREAKING:** Integrate PollingController mixin with GasFeeController ([#1673](https://github.com/MetaMask/core/pull/1673), [#1748](https://github.com/MetaMask/core/pull/1748))
```

File: packages/network-controller/CHANGELOG.md
Comment (mcmire) [inline packages/network-controller/CHANGELOG.md:11]: Instead of repeating the name of the PR, what do you think about listing the changes that are relevant to consumers of this package? It seems like we would want something like:

``` suggestion
### Added
- Add `NetworkController:getEIP1559Compatibility` controller action ([#1673](https://github.com/MetaMask/core/pull/1673))

### Changed
- **BREAKING:** Rename `get1555CompatibilityWithNetworkClientId` to `get1559CompatibilityWithNetworkClientId` ([#1673](https://github.com/MetaMask/core/pull/1673))
```

File: packages/polling-controller/CHANGELOG.md
Comment (mcmire) [inline packages/polling-controller/CHANGELOG.md:9]: 0.x as a version denotes a "pre-production" release which is highly subject to change, and I feel that we would only do that if we are not confident about the readiness of the code. But, if we _do_ feel confident, what are our thoughts on using 1.0.0 instead of 0.0.1?

Comment (adonesky1) [reply packages/polling-controller/CHANGELOG.md:9]: So this package is pre-production in the sense that I don't expect we will be using it's functionality just yet. What do we think about versioning here @shanejonas @Gudahtt

Comment (shanejonas) [reply packages/polling-controller/CHANGELOG.md:9]: hm yea technically it wont be used, I do like having a pre 1.0.0 release since we might change some of the interface before a true 1.0.0, like we did already with `startPollingByNetworkClientId` method name.

Comment (adonesky1) [reply packages/polling-controller/CHANGELOG.md:9]: Yeah it feels like there may still be some changes to this mixin/abstract class needed before it goes to prod.

Comment (mcmire) [reply packages/polling-controller/CHANGELOG.md:9]: Oh okay. What do we feel about 0.1.0 then? Where the second number sort of takes the place of major and the third number takes the place of minor.

Comment (adonesky1) [reply packages/polling-controller/CHANGELOG.md:9]: Yep sounds good

**Comment by adonesky1** [comment]
I'm currently rebasing and updating this release

File: packages/approval-controller/CHANGELOG.md
Comment (mcmire) [inline packages/approval-controller/CHANGELOG.md:12]: I'm currently checking whether this will cause breaking changes, stand by...

**Review by mcmire — CHANGES_REQUESTED** [review]
Marking this as "request changes" until I can verify that the `rpc-errors` -> `@metamask/rpc-errors` upgrade doesn't cause breaking changes.

File: packages/approval-controller/CHANGELOG.md
Comment (mcmire) [reply packages/approval-controller/CHANGELOG.md:12]: Hmm... so it does seem this causes breaking changes. Namely, if you are still using `eth-rpc-errors` to build errors and pass them to `approval-controller`, then you need to start using `@metamask/rpc-errors` instead, otherwise you will get a type error:

```
import {ControllerMessenger} from "@metamask/base-controller";
import {ApprovalController} from "@metamask/approval-controller";
import {ethErrors} from "eth-rpc-errors";

function main() {
  const messenger = new ControllerMessenger();
  const restrictedMessenger = messenger.getRestricted({
    name: "ApprovalController",
    allowedActions: [],
    allowedEvents: [],
  });
  const approvalController = new ApprovalController({
    messenger: restrictedMessenger,
    showApprovalRequest: () => {
      // whatever
    },
    state: {},
    typesExcludedFromRateLimiting: [],
  });
  const rejectionError = ethErrors.rpc.invalidRequest({
    data: new Date()
  });
  // TypeScript error:
  // Type 'Date | undefined' is not assignable to 'DataWithOptionalCause | undefined'.
  approvalController.clear(rejectionError);
}
```

So maybe we should bump this package to 4.0.0 and leave a note here on how to upgrade?

Comment (adonesky1) [reply packages/approval-controller/CHANGELOG.md:12]: 🤷‍♂️

File: packages/gas-fee-controller/CHANGELOG.md
Comment (mcmire) [inline packages/gas-fee-controller/CHANGELOG.md:12]: Can we list the breaking changes here? I'm also noticing some additions:

``` suggestion
## Added
- Add optional `gasFeeEstimatesByChainId` property to GasFeeController state ([#1673](https://github.com/MetaMask/core/pull/1673)
- Add dependency on `@metamask/polling-controller` ([#1748])(https://github.com/MetaMask/core/pull/1748))

## Changed
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- **BREAKING:** Messenger must allow controller actions `NetworkController:getNetworkClientById` and `NetworkController:getEIP1559Compatibility` ([#1673](https://github.com/MetaMask/core/pull/1673)
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `@metamask/controller-utils` to ^5.0.2
- Bump dependency and peer dependency on `@metamask/network-controller` to ^14.0.0
```

File: packages/keyring-controller/CHANGELOG.md
Comment (mcmire) [inline packages/keyring-controller/CHANGELOG.md:11]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `@metamask/message-manager` to ^7.3.5

File: packages/message-manager/CHANGELOG.md
Comment (mcmire) [inline packages/message-manager/CHANGELOG.md:11]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `metamask/controller-utils` to ^5.0.2

File: packages/name-controller/CHANGELOG.md
Comment (mcmire) [inline packages/name-controller/CHANGELOG.md:11]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency on `@metamask/base-controller` to ^3.2.3
```

File: packages/network-controller/CHANGELOG.md
Comment (mcmire) [inline packages/network-controller/CHANGELOG.md:15]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `@metamask/controller-utils` to ^5.0.2

File: packages/notification-controller/CHANGELOG.md
Comment (mcmire) [inline packages/notification-controller/CHANGELOG.md:11]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency on `@metamask/base-controller` to ^3.2.3
```

File: packages/permission-controller/CHANGELOG.md
Comment (mcmire) [inline packages/permission-controller/CHANGELOG.md:14]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency and peer dependency on `@metamask/approval-controller` to ^3.5.3
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `@metamask/controller-utils` to ^5.0.2

File: packages/polling-controller/CHANGELOG.md
Comment (mcmire) [inline packages/polling-controller/CHANGELOG.md:11]: Hmm... it seems that 0.0.0 got released somehow. We should probably revoke that release.

In any case 0.1.0 seems like the proper initial release, yeah? Maybe we should denote it as such.

``` suggestion
### Added
- Initial release
```

File: packages/signature-controller/CHANGELOG.md
Comment (mcmire) [inline packages/signature-controller/CHANGELOG.md:11]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency and peer dependency on `@metamask/approval-controller` to ^3.5.3
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `@metamask/controller-utils` to 5.0.2
- Bump dependency on `@metamask/message-manager` to ^7.3.5

File: packages/transaction-controller/CHANGELOG.md
Comment (mcmire) [inline packages/transaction-controller/CHANGELOG.md:11]: Also:

``` suggestion
- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
- Bump dependency and peer dependency on `@metamask/approval-controller` to ^3.5.3
- Bump dependency on `@metamask/base-controller` to ^3.2.3
- Bump dependency on `@metamask/controller-utils` to ^5.0.2
- Bump dependency and peer dependency on `@metamask/network-controller` to ^14.0.0

**Review by mcmire — COMMENTED** [review]
Okay, I've looked through this whole PR. One thing to keep in mind is that since we're publishing new versions of base packages like `base-controller` and `controller-utils`, and since dependencies on those packages within the monorepo are also being bumped, those dependency bumps need to be documented in changelog updates. We will figure out a way to automate this so it doesn't have to be done by hand, but in the meantime I've made a bunch of suggestions.

**Comment by mcmire** [comment]
I've gone ahead and accepted the minor suggestions I made above. For the only two that are left, I'm not able to resolve those myself. As they are non-blocking, though, I've taken off my rejection earlier.

File: packages/message-manager/CHANGELOG.md
Comment (mcmire) [reply packages/message-manager/CHANGELOG.md:16]: What do you think @OGPoyraz?

**Comment by mcmire** [comment]
I'll check back tomorrow on this, but for anyone who gets to this before I do, feel free to merge this.

File: packages/message-manager/CHANGELOG.md
Comment (OGPoyraz) [reply packages/message-manager/CHANGELOG.md:16]: Sure make sense, thanks for suggestion 👍

**Review by matthewwalsh0 — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by mcmire — DISMISSED** [review]
Looks like we need to update changelogs again. I will do another review and make changes as necessary.

File: packages/gas-fee-controller/CHANGELOG.md
Comment (mcmire) [reply packages/gas-fee-controller/CHANGELOG.md:12]: I'm just going to go ahead and accept the above, as it makes more sense to me.

**Comment by mcmire** [comment]
^ This is now ready for another review, hopefully we can get this merged soon.

File: packages/transaction-controller/CHANGELOG.md
Comment (adonesky1) [inline packages/transaction-controller/CHANGELOG.md:10]: Do we need to include the changelogs from https://github.com/MetaMask/core/pull/1725?

Comment (adonesky1) [reply packages/transaction-controller/CHANGELOG.md:10]: oh lol theyre there whoops

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Comment by mcmire** [comment]
Hmm, looks like we need a review from `snaps-devs`. Requesting...

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**
